include/com/centreon/exceptions/config.hh

```diff
@@ -18,23 +18,30 @@
 
 #ifndef CCB_EXCEPTIONS_CONFIG_HH
 #define CCB_EXCEPTIONS_CONFIG_HH
+#include <fmt/format.h>
 
-#include "com/centreon/broker/exceptions/msg.hh"
-#include "com/centreon/broker/namespace.hh"
+#include <exception>
+#include <string>
 
-CCB_BEGIN()
+#include "com/centreon/exceptions/msg_fmt.hh"
+#include "com/centreon/namespace.hh"
+
+CC_BEGIN()
 
 namespace exceptions {
 /**
- *  @class config config.hh "com/centreon/broker/exceptions/config.hh"
+ *  @class config config.hh "com/centreon/exceptions/config.hh"
  *  @brief Configuration exception.
  *
  *  Such exceptions are thrown in case of configuration errors.
  */
-class config : public msg {
+class config : public msg_fmt {
  public:
-  config() = default;
-  config(const config&) = default;
+  template <typename... Args>
+  explicit config(std::string const& str, const Args&... args)
+      : msg_fmt(fmt::format(str, args...)) {}
+
```

Comment:
I think you can remove the call to fmt::format here
and just call msg_fmt(str, args...)